packages/rum-core/test/testSetupBuilder.ts

```diff
@@ -78,6 +81,7 @@ export function setup(): TestSetupBuilder {
   let sessionManager: RumSessionManager = createRumSessionManagerMock().setId('1234')
   const lifeCycle = new LifeCycle()
   const domMutationObservable = new Observable<void>()
+  const performanceResourceObservable = new Observable<RumPerformanceResourceTiming[]>()
```

Comment:
**💬 suggestion:** ‏I think we should migrate away of the `SetupBuilder`, because it's not so relevent anymore. Maybe you could create observables in the specs that need it instead?

Reply:
Out of curiosity, what makes you feel it is not relevant anymore. (I also agree we should move away from it :)

Reply:
* It's designed to test high level modules like `startRum`, where it makes kind of sense, but used in lower level components where it brings more complexity than needed as it's not focused enough. Just creating the dependencies / running the function in the spec is generally enough. When this becomes too verbose, introducing a `xxxWithDefaults` function helps ([ex](https://github.com/DataDog/browser-sdk/blob/6102f81637a73a3d825bde5f2c896a67fa90d66a/packages/rum-core/src/domain/rumSessionManager.spec.ts#L217)). This makes the spec self contained and we avoid one indirection.

* Now that we have `registerCleanupTask`, it loses one of the main benefit/use-case, which was conditionally starting something in a test case knowing that it'll be cleaned up with `setup.cleanup()` anyway. For example, I'd like to improve `mockClock` by using `registerCleanupTask`, so we have a single way to mock the clock that can be used across the SDK, not only in RUM 

* We'll need to figure out how we could keep RUM event validation though

Reply:
I removed it from the builder